packages/dds/tree/src/test/shared-tree/fuzzEditGenerator.ts

```diff
@@ -0,0 +1,331 @@
+    fuzzType: "insert";
+    parent: UpPath | undefined;
+    field: FieldKey;
+    index: number;
+    value: number;
+}
+
+export interface FuzzDelete {
+    fuzzType: "delete";
+    path: UpPath | undefined;
+}
+
+export interface FuzzSetPayload {
+    fuzzType: "setPayload";
+    path: UpPath | undefined;
+    value: number;
+}
+
+export type FuzzChange = FuzzInsert | FuzzDelete | FuzzSetPayload;
+
+export interface Synchronize {
+    type: "synchronize";
+}
+
+export interface FuzzTestState extends BaseFuzzTestState {
+    testTreeProvider: ITestTreeProvider;
+    numberOfEdits: number;
+}
+
+export interface TreeContext {
+    treeIndex: number;
+}
+
+export const makeEditGenerator = (): AsyncGenerator<Operation, FuzzTestState> => {
+    type EditState = FuzzTestState & TreeContext;
+    async function insertGenerator(state: EditState): Promise<FuzzInsert> {
+        const trees = state.testTreeProvider.trees;
+        const tree = trees[state.treeIndex];
+
+        // generate edit for that specific tree
+        const {
+            parent: path,
+            field: nodeField,
+            index: nodeIndex,
+        } = getRandomNodePosition(tree, state.random);
+        const insert: FuzzInsert = {
+            fuzzType: "insert",
+            parent: path,
+            field: nodeField,
+            index: nodeIndex,
+            value: state.random.integer(Number.MIN_SAFE_INTEGER, Number.MIN_SAFE_INTEGER),
+        };
+        return insert;
+    }
+
+    async function deleteGenerator(state: EditState): Promise<FuzzDelete> {
+        const trees = state.testTreeProvider.trees;
+        const tree = trees[state.treeIndex];
+        // generate edit for that specific tree
+        const path = containsAtLeastOneNode(tree)
+            ? getExistingRandomNodePosition(tree, state.random)
+            : undefined;
+        return {
+            fuzzType: "delete",
+            path,
+        };
+    }
+
+    async function setPayloadGenerator(state: EditState): Promise<FuzzSetPayload> {
+        const trees = state.testTreeProvider.trees;
+        const tree = trees[state.treeIndex];
+        // generate edit for that specific tree
+        const path = containsAtLeastOneNode(tree)
+            ? getExistingRandomNodePosition(tree, state.random)
+            : undefined;
+        return {
+            fuzzType: "setPayload",
+            path,
+            value: state.random.integer(Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER),
+        };
+    }
+
+    /**
+     * currently the acceptance conditions are based on some arbitrary number of edits.
+     * This should be changed to the max number of nodes allowed in tree.
+     * TODO: create helper function to get number of nodes in tree and use that to set acceptance condition
+     * */
+    const baseEditGenerator = createWeightedAsyncGenerator<FuzzChange, EditState>([
+        [insertGenerator, 5],
+        [deleteGenerator, 1],
+        [setPayloadGenerator, 1],
```

Comment:
You could use `containsAtLeastOneNode` as conditionals for these two generators.
This way the `path` fields won't need to tolerate undefined in `FuzzDelete` and `FuzzSetPayload`.